src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs

```diff
@@ -488,9 +488,10 @@ private bool MatchesSubtype(MediaType set)
             }
             else
             {
-                // The set has no suffix, so we're just looking for an exact match (which means that if 'this'
-                // has a suffix, it won't match).
-                return set.SubType.Equals(SubType, StringComparison.OrdinalIgnoreCase);
+                // If this subtype matches either the subtype or suffix of the set,
```

Comment:
`If this subtype or suffix matches the subtype of the set,`

Not sure whether this comment is backward in aspnet/HttpAbstractions too.